CONTRIBUTING.md

```diff
@@ -1,61 +1,101 @@
+#### Clone your fork
 
-    cd modular-docs
-    git remote add upstream git@github.com:redhat-documentation/modular-docs.git
+2. **Clone** your fork to your local machine:
 
-STEP 5 - Verify that you have two remotes, `origin` and `upstream` and check the status of your project:
+   ```bash
+   git clone git@github.com:<username>/modular-docs.git
+   ```
 
-    git remote -v
-    git status
+   For more information, see GitHub's [Cloning your forked repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#cloning-your-forked-repository) topic.
 
-## Contributing Changes
+#### Configure an upstream remote
 
-STEP 1 - Create a new branch:
+3. **Configure an upstream remote** to sync with the original repository:
 
-    git checkout -b <new_branch_name>
+   ```bash
+   cd modular-docs
+   git remote add upstream git@github.com:redhat-documentation/modular-docs.git
+   ```
 
-Example:
+   For more information, see GitHub's [Configuring git to sync your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#configuring-git-to-sync-your-fork-with-the-upstream-repository) topic.
 
-    git checkout -b new_edits
+#### Verify the remotes
 
-STEP 2 - Add content or make edits.
+4. **Verify the remotes** and check your project's status:
 
-STEP 3 - Add and commit your changes:
+   ```bash
+   git remote -v
+   git status
+   ```
 
-    git add <path_to_file>
-    git commit -m "<place_your_comments_here>"
+## Making contributions
 
- Example:
+### Stay updated
 
-    git add modular-doc-manual/master.adoc
-    git commit -m -s "Updated the master.adoc file"
+* **Prepare your main branch**:
 
-STEP 4 - Push changes to the remote GitHub repository:
+   ```bash
+   git checkout main
+   git fetch upstream main
+   git rebase upstream/main  
+   git checkout -b <new_branch_name>
+   ```
 
-    git push origin <new_branch_name>
+   Example:
 
-Example:
+   ```bash
+   git checkout -b new_edits
+   ```
 
-    git push origin new_edits
+### Make your changes
 
-STEP 5 - Create a new pull request from the GitHub web interface.
+1. **Edit content** or add new material as needed.
 
-STEP 6 - Everyone on the project team will review the merge request and add comments in GitHub. This review process will be open for one week from the day the merge request was submitted. If the merge request is still being actively discussed beyond the one week timeframe, then the merge request will stay open. Once the merge request discussion is resolved, the merge request will be NACK'd or ACK'd based on the comments given.  If no comments are given after a week, then the merge request will be ACK'd.
+2. **Stage, commit, and push** your updates:
+
+   ```bash
+   git add <path_to_file>
+   git commit -m "<your_commit_message_here>"
+   git push -f origin HEAD
```

Comment:
I'm not sure about this. I would consider to simplify the first two commands by merging them into one:
`git commit -am "<your_commit_message_here>"`
Then, I believe that in our case, a simple `git push` is enough (when you locally configured Git to automatically set up remote), or:
` git push --set-upstream origin <new_branch_name>`.
Or do we need to force push?

Reply:
I believe `git commit -am "<your_commit_message_here>"` only operates on already-tracked files. For untracked files, the user must still use `git add <filename>` before commiting the changes.

Reply:
> when you locally configured Git to automatically set up remote

I'm not sure we can rely on this being true in most cases.

Reply:
> `git push --set-upstream origin <new_branch_name>`

also works but is longer than `git push origin HEAD`, which I beleive most writers are familiar with.

Reply:
> Or do we need to force push?

We don't need to force push the first time, but it's harmless. I've removed it.